src/helpers/ClauseResultsHelpers.ts

```diff
@@ -49,7 +49,37 @@ export function findAllLocalIdsInStatementByName(libraryElm: ELM, statementName:
       clause.isUnsupported = true;
     }
   }
-  return localIds;
+
+  // find all localids in the annotation
+  const allAnnotatedIds = findAnnotationLocalIds(statement?.annotation);
+  // filter out local ids that aren't in the annotation
+  const annotatedLocalIds: { [key: string]: any } = {};
+  for (const [key, value] of Object.entries(localIds)) {
+    if (allAnnotatedIds.includes(key)) {
+      annotatedLocalIds[key] = value;
+    }
+  }
+  return annotatedLocalIds;
+}
+
+/**
+ * Recursively finds just localIds that are in an annotation structure by pulling out all "r:"-keyed values
+ * @public
+ * @param {object} annotation - all or a subset of the annotation structure to search
+ * @return {Array} List of local ids in the annotation.
+ */
+function findAnnotationLocalIds(annotation: any): any[] {
```

Comment:
This is pretty clever even if a bit hard to read. I think it could be specific about the return type being `string[]` though. I realize the existing code around this doesn't have proper types, but I think this new code could start to use something other than `any`.

Reply:
Legit- I'll give it a shot!